packages/protocol/contracts/team/airdrop/ERC20Airdrop.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import { BaseAirdrop } from "./BaseAirdrop.sol";
+import { Proxied } from "../../common/Proxied.sol";
+
+/// @dev This shall be implemented by the vaults helping with the airdrop.
+interface IERC20 {
+    function transferFrom(
+        address from,
+        address to,
+        uint256 value
+    )
+        external
+        returns (bool);
+}
+
+/// @title ERC20Airdrop
+/// Contract for managing Taiko token airdrop for eligible users
+contract ERC20Airdrop is BaseAirdrop {
+    struct ERC20InputParams {
+        bytes32[] merkleProof; // Format is given (see OZ code)
+        uint256 allowance;
+    }
+
+    event UserClaimed(address indexed user, uint256 amount);
+
+    /// @notice Claim airdrop
+    /// @param inputData Encoded data containing merkle proof and related
+    /// variables (amount, contract, etc.)
+    function claim(bytes calldata inputData) external virtual override {
+        if (merkleRoot == 0x0) {
+            revert CLAIM_NOT_STARTED();
+        }
+
+        ERC20InputParams memory input =
+            abi.decode(inputData, (ERC20InputParams));
+
+        // Create the leafs - a leaf consist of the 2 data encoded (address and
+        // allowance). See claimList.ts and buildMerkleTree.ts for more info !
+        bytes32 leaf = keccak256(abi.encodePacked(msg.sender, input.allowance));
```

Comment:
Would allow people to claim for different addresses as well instead of requiring `msg.sender` to do it itself.

Reply:
I see the point, but from a practical standpoint, how you “chose” from the (example) 10K database ?
You (Bob) goes to the website, and via the UI you cherry-pick Alice address to claim for her ? 🤔 would not it make sense that everyone claims himself/herself ?

Reply:
Yeah you check for eligibility for all your addresses and then claim them using an address that is most convenient. For example I have an address I just use for doing small transactions that is easier to use than other addresses that may require more work (hardware wallet/smart wallet). It can also be safer in those cases because you don't have to do any transaction from those wallets that contain more wallets so have to be more careful when doing transactions for those.

Reply:
Sure, but it would mean everyone could claim (and check eligibility) for everyone.. if which is OK with you, then it is OK with me :) 
Another Q for the UI then. So you go to the platform, and you connect your wallet ?  Or just there is a sinple input text form, where you can initiate a query per specific address ?
And if address is eligible, then you can request for proofs.. and can claim for anybody ?